### ISSUE TITLE
Linking error for compiling example of Triangulation_2

### ISSUE THREAD

**Issue by kylelo** [CGAL/cgal#4511]
## Issue Details

I followed the instruction [here](https://cgal.geometryfactory.com/CGAL/doc/master/Manual/usage.html#usage_introduction) to compile the example of Triangulation_2. I have installed all prerequisites (cmake, boost, and cgal) using brew. However, the linking error still showed up as 

```
kyles-MacBook-Pro-2:Triangulation_2 kylelo$ make
[  2%] Building CXX object CMakeFiles/regular.dir/regular.cpp.o
[  4%] Linking CXX executable regular
Undefined symbols for architecture x86_64:
  "CGAL::get_mode(std::__1::basic_ios<char, std::__1::char_traits<char> >&)", referenced from:
      std::__1::basic_istream<char, std::__1::char_traits<char> >& CGAL::extract<CGAL::Epick>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, CGAL::Weighted_point_2<CGAL::Epick>&, CGAL::Cartesian_tag const&) in regular.cpp.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& CGAL::insert<CGAL::Epick>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, CGAL::Weighted_point_2<CGAL::Epick> const&, CGAL::Cartesian_tag const&) in regular.cpp.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& CGAL::insert<CGAL::Epick>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, CGAL::Point_2<CGAL::Epick> const&, CGAL::Cartesian_tag const&) in regular.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [regular] Error 1
make[1]: *** [CMakeFiles/regular.dir/all] Error 2
make: *** [all] Error 2
```

## Source Code

/CGAL-5.0.1/examples/Triangulation_2

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Mac0S 10.15.2
* Compiler: AppleClang 11.0.0.11000033
* Release or debug mode: Release
* Specific flags used (if any): -DCMAKE_BUILD_TYPE=Release
* CGAL version: 5.0.1
* Boost version: 1.72.0

## Solution

I found this problem is caused by linking issue of cgal library on the MacOS.
The following command fixed the issue
>  brew link cgal

Recommend to add this as a note on 2. prerequisites for MacOS users who need it.

**Comment by lrineau** [comment]
@maxGimeno @danston, can you have a look together what a call to `brew link cgal` does, and in which scenario it can be required?

**Comment by danston** [comment]
@lrineau Sure, I will try to compile these examples today and see if we actually need to do this linking. In fact, what linking does is very simple, it links the CGAL folder from the brew cellar to the usr/local so that it could be found globally if I am not mistaken, but again I will check.

**Comment by lrineau** [comment]
@kylelo That is strange you needed to call `brew link cgal`. Do you have any idea why brew has not called that command automatically after having installed CGAL?

**Comment by danston** [comment]
Hey @kylelo. I have just tested Triangulation 2 examples in the Release mode on my Mac OS Mojave 10.14.6 with the boost 1.72.0, cmake 3.16.4, CGAL 5.0.1, Apple clang version 11.0.0 (clang-1100.0.33.16) and the latest brew version. Basically what I did is the following:

```bash
brew update
brew upgrade
brew cleanup
brew install cmake
brew install boost
brew install gmp
brew install mpfr
brew install cgal
cd /Users/username/Downloads/CGAL-5.0.1/examples/Triangulation_2
cmake -DCMAKE_BUILD_TYPE=Release .
make
```

I have successfully compiled all the examples and managed to run all of them. I do not think that 

```bash
brew link cgal
``` 

is a good idea! brew link is not necessary because CGAL should be linked (and seen later globally) automatically by brew when installed. 

What could cause your problem is either a wrong configuration of the brew or its packages or something wrong in your OS configuration. If you manually linked and unlinked something before, then when installing CGAL brew could decide to leave your CGAL package unlinked too, because otherwise it could brake something else more important. Since you also have a new version of Mac OS, it might be that brew is not fully supported/patched on it yet, though I doubt it!

**Comment by kylelo** [comment]
> @kylelo That is strange you needed to call `brew link cgal`. Do you have any idea why brew has not called that command automatically after having installed CGAL?

> Hey @kylelo. I have just tested Triangulation 2 examples in the Release mode on my Mac OS Mojave 10.14.6 with the boost 1.72.0, cmake 3.16.4, CGAL 5.0.1, Apple clang version 11.0.0 (clang-1100.0.33.16) and the latest brew version. Basically what I did is the following:
> 
> ```shell
> brew update
> brew upgrade
> brew cleanup
> brew install cmake
> brew install boost
> brew install gmp
> brew install mpfr
> brew install cgal
> cd /Users/username/Downloads/CGAL-5.0.1/examples/Triangulation_2
> cmake -DCMAKE_BUILD_TYPE=Release .
> make
> ```
> 
> I have successfully compiled all the examples and managed to run all of them. I do not think that
> 
> ```shell
> brew link cgal
> ```
> 
> is a good idea! brew link is not necessary because CGAL should be linked (and seen later globally) automatically by brew when installed.
> 
> What could cause your problem is either a wrong configuration of the brew or its packages or something wrong in your OS configuration. If you manually linked and unlinked something before, then when installing CGAL brew could decide to leave your CGAL package unlinked too, because otherwise it could brake something else more important. Since you also have a new version of Mac OS, it might be that brew is not fully supported/patched on it yet, though I doubt it!

Thank you for the confirmation. The reason might be I have tried to compile cgal a year ago. After removed the repo, I didn't remove the cgal lib on my mac. Using brew to install cgal again caused the linking error.

**Comment by danston** [comment]
> Thank you for the confirmation. The reason might be I have tried to compile cgal a year ago. After removed the repo, I didn't remove the cgal lib on my mac. Using brew to install cgal again caused the linking error.

Ok, I see.

**Comment by lrineau** [comment]
#4516 is now merged in the branch for 5.0.2. See https://github.com/CGAL/cgal/blob/bf31ed804de2800528f9194d1a3dbf992de5c187/INSTALL.md.
